modules/swagger-codegen/src/main/resources/swift3/AlamofireImplementations.mustache

```diff
@@ -77,7 +77,18 @@ open class AlamofireRequestBuilder<T>: RequestBuilder<T> {
      configuration (e.g. to override the cache policy).
      */
     open func makeRequest(manager: SessionManager, method: HTTPMethod, encoding: ParameterEncoding, headers: [String:String]) -> DataRequest {
-        return manager.request(URLString, method: method, parameters: parameters, encoding: encoding, headers: headers)
+        if isBody {
```

Comment:
I might be missing something but do you need this if block?  if you change parameters to an Any? like you have instead of a dict, won't ParameterEncoding.encode() do exactly what you have here?

Reply:
hmmm... now that I'm looking at how the ParameterEncoding and Manager works I think something like this is cleaner. I think the function show also changed to throws so we can handle errors better. Then change the execute to use do/catch and generate a proper error.

```
let originalRequest = try URLRequest(url: URLString, method: method, headers: headers)
if let encoding = encoding as? JSONEncoding {
    let encodedRequest = try encoding.encode(originalRequest, withJSONObject: parameters)
    return manager.request(encodedRequest)
} else {
    let encodedRequest = try encoding.encode(originalRequest, with: parameters as? Parameters)
    return manager.request(encodedRequest)
}
```

The main problem with Alamofire is that parameter is defined as `Parameters` which is a typedef of [String: Any]. If you want to specify anything else they make you do everything your self.

Let me try making these changes and run some tests.